nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -638,8 +637,12 @@ public void setUpClient(final ProcessContext context) throws IOException, Unreco
         }
 
         // Set timeouts
-        okHttpClientBuilder.connectTimeout((context.getProperty(PROP_CONNECT_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue()), TimeUnit.MILLISECONDS);
-        okHttpClientBuilder.readTimeout(context.getProperty(PROP_READ_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue(), TimeUnit.MILLISECONDS);
+        okHttpClientBuilder.connectTimeout((context.getProperty(PROP_CONNECT_TIMEOUT).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS).intValue()), TimeUnit.MILLISECONDS);
+
+        if (context.getProperty(PROP_READ_TIMEOUT).isExpressionLanguagePresent())
+            okHttpClientBuilder.addInterceptor(new TimeoutInterceptor());
```

Comment:
Can you explain why an interceptor is added here rather than passing the configured value to the builder?

Reply:
This interceptor set timeout for every request(if timeout is a expression language). Timeout value is different for every request so i think there is no reason to initialize.

Reply:
See my comment on `newBuilder()` bewlow. I think it's right that it's not sufficient to set it on the primary builder b/c you want to use per-flowfile attributes, but I think the static timeout + an interceptor is problematic.